src/environment.cpp

```diff
@@ -112,10 +112,52 @@ inline static bool isPointableNode(const MapNode &n,
 	       (liquids_pointable && features.isLiquid());
 }
 
+inline static void updateMaxOverlap(f32 overlap, f32 extent, f32& max_overlap, f32& max_overlap_extent)
+{
+	if (overlap > 0 && (max_overlap_extent == 0 || overlap * max_overlap_extent > extent * max_overlap)) {
+		max_overlap = overlap;
+		max_overlap_extent = extent;
+	}
+}
+
 void Environment::continueRaycast(RaycastState *state, PointedThing *result)
 {
 	const NodeDefManager *nodedef = getMap().getNodeDefManager();
 	if (state->m_initialization_needed) {
+		// Check boundaries
+		const float limit = MAX_MAP_GENERATION_LIMIT * BS;
+		if (!std::isfinite(state->m_shootline.start.X) ||
+			!std::isfinite(state->m_shootline.start.Y) ||
+			!std::isfinite(state->m_shootline.start.Z) ||
+			!std::isfinite(state->m_shootline.end.X) ||
+			!std::isfinite(state->m_shootline.end.Y) ||
+			!std::isfinite(state->m_shootline.end.Z) ||
+			std::abs(state->m_shootline.start.X) > limit ||
+			std::abs(state->m_shootline.start.Y) > limit ||
+			std::abs(state->m_shootline.start.Z) > limit ||
+			std::abs(state->m_shootline.end.X) > 1000000 * limit ||
```

Comment:
1000000? what? why?

Reply:
Where would this happen? `pow(32000 * 10 * 2, 2)` for an assumed worst-case scenario is still well within bounds of floating point values.

Reply:
Do you assume end <`pow(32000 * 10 * 2, 2)`? if so, why?

Reply:
Mods work mostly within bounds and raycasts are limited by the loaded block range. Hence a distance of 100m is already unreliable, let alone `1000 * BS` outside the bounds. That would very likely be a calculation error. Garbage in results in garbage out.
Granted, from a technical perspective of floating point arithmetic, yours is an acceptable sane solution. Anyway, an in-line comment would help a lot to understand this magic number.

Reply:
At least it wouldn't freeze the server, right?

Reply:
No more magic numbers